src/http/StatusLine.h

```diff
@@ -44,7 +44,7 @@ class StatusLine
     /// retrieve the status code for this status line
     Http::StatusCode status() const { return status_; }
 
-    /// retrieve the reason string for this status line
+    /// the reason-phrase string; never nil
     const char *reason() const;
```

Comment:
I adjusted the existing description to document the never-nil guarantee used by the new caller of this method. This PR does not change how this method works. A lot can be polished in this and related classes, but this PR is not about that.